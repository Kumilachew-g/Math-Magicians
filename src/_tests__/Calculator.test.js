import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import Calculator from '../components/Calculator';

describe('Calculator page component', () => {
  it('should return the sum of two number form UI', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByText('21')).toBeTruthy();
  });
});
