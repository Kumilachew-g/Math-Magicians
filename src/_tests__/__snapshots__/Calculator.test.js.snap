// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders page correctly 1`] = `
<section
  className="calculator-container"
>
  <h1>
    Let's do some math!
  </h1>
  <div
    className="calculator-section"
  >
    <div
      className="output-section"
    >
      0
    </div>
    <div
      className="digits"
    >
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="buttons operator-signs"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="buttons operator-signs"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="buttons operator-signs"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="buttons operator-signs"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
    </div>
    <span
      className="bottom"
    >
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className="buttons operator-signs"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </span>
  </div>
</section>
`;
